torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -772,8 +797,8 @@ def call_function(
             *(arg.as_proxy() for arg in args),
             *(arg for arg in body_lifted_freevars.keys()),
         )
-        r = body_r.as_proxy().node.meta["example_value"]
-        example_value = r
+
+        example_value = _unpack(tx, body_r)
```

Comment:
We actually already handle this non-tuple returns for `wrap` and checkpoint: https://github.com/pytorch/pytorch/blob/053654b9cf9ccb7f4f1746e7d324d694ce98f9f1/torch/_dynamo/variables/higher_order_ops.py#L826-L831 . It looks like we never brought the improvement over to autograd.Function. We should probably be consistent with how we solve the problem. 

I'm not completely sure what the difference is between `_unpack` and what we do with wrap, but we should settle on one and use it everywhere

Reply:
They are subtly different - but I think we can pick one for our purposes. I just didn't know this existed! Let me try reusing this. 

> It looks like we never brought the improvement over to autograd.Function

Or functorch, it seems?